### PR TITLE
Added the recv_tos socket option

### PR DESCRIPTION
**Pull request by JoNil**

### CODE REVIEW

File: src/sys/unix.rs [hunk 1]
```diff
@@ -83,6 +83,13 @@ pub(crate) use libc::IP_HDRINCL;
     target_os = "solaris",
     target_os = "illumos",
 )))]
+pub(crate) use libc::IP_RECVTOS;
```
Comment (Thomasdezeeuw) [inline src/sys/unix.rs:86]: I think this `cfg` is the same as the one for `IP_TOS`, can you merge the two.

**Review by Thomasdezeeuw — CHANGES_REQUESTED** [review]
The addition sounds good, but this needs tests and Windows support (see https://docs.microsoft.com/en-us/windows/win32/winsock/ipproto-ip-socket-options).

**Comment by JoNil** [comment]
I didn't find the IP_RECVTOS symbol in winapi-rs. Does anyone know if it is supported there and is just not exposed?

**Comment by Thomasdezeeuw** [comment]
> I didn't find the IP_RECVTOS symbol in winapi-rs. Does anyone know if it is supported there and is just not exposed?

I can't find it either.  Perhaps we should move to `windows_sys` (it has it: https://docs.rs/windows-sys/latest/windows_sys/Win32/Networking/WinSock/constant.IP_RECVTOS.html), but that's a larger topic. For now you can remove Window support and we'll open issue to add it later.

**Comment by JoNil** [comment]
In the last commit i added windows the the cfg so that it is not included :)

**Review by Thomasdezeeuw — APPROVED** [review]

**Comment by Thomasdezeeuw** [comment]
Thanks @JoNil.

**Pull request merged**